tests/playwright/src/model/pages/create-machine-page.ts

```diff
@@ -0,0 +1,137 @@
+    this.podmanMachineConfiguration = this.page.getByRole('form', { name: 'Properties Information' });
+    this.podmanMachineName = this.podmanMachineConfiguration.getByRole('textbox', { name: 'Name' });
+    this.imagePathBox = this.podmanMachineConfiguration.getByRole('textbox', { name: 'Image Path (Optional) ' });
+    this.browseImagesButton = this.podmanMachineConfiguration.getByRole('button', {
+      name: 'button-Image Path (Optional)',
+    });
+    this.podmanMachineCPUs = this.podmanMachineConfiguration.getByRole('slider', { name: 'CPU(s)' });
+    this.podmanMachineMemory = this.podmanMachineConfiguration.getByRole('slider', { name: 'Memory' });
+    this.podmanMachineDiskSize = this.podmanMachineConfiguration.getByRole('slider', { name: 'Disk size' });
+    this.rootPriviledgesCheckbox = this.podmanMachineConfiguration.getByRole('checkbox', {
+      name: 'Machine with root privileges',
+    });
+    this.userModeNetworkingCheckbox = this.podmanMachineConfiguration.getByRole('checkbox', {
+      name: 'User mode networking',
+    });
+    this.startNowCheckbox = this.podmanMachineConfiguration.getByRole('checkbox', { name: 'Start the machine now' });
+    this.closeButton = this.page.getByRole('button', { name: 'Close' });
+    this.createMachineButton = this.page.getByRole('button', { name: 'Create' });
+  }
+
+  async createMachine(
+    machineName: string,
+    { isRootful = true, enableUserNet = false, startNow = true, setAsDefault = true },
+  ): Promise<ResourcesPage> {
+    await playExpect(this.podmanMachineConfiguration).toBeVisible();
+    await this.podmanMachineName.fill(machineName);
+
+    await this.ensureCheckboxState(isRootful, this.rootPriviledgesCheckbox);
+    await this.ensureCheckboxState(enableUserNet, this.userModeNetworkingCheckbox);
+    await this.ensureCheckboxState(startNow, this.startNowCheckbox);
+
+    await this.createMachineButton.click();
+    await this.page.waitForTimeout(60_000);
+
+    const successfulCreationMessage = this.page.getByText('Successful operation');
+    const goBackToResourcesButton = this.page.getByRole('button', { name: 'Go back to resources' });
+
+    await delay(5000);
+    await this.handleConnectionDialog(machineName, setAsDefault);
+
+    await playExpect(successfulCreationMessage).toBeVisible({ timeout: 10_000 });
+    await playExpect(goBackToResourcesButton).toBeVisible();
+    await goBackToResourcesButton.click();
+
+    return new ResourcesPage(this.page);
+  }
+
+  async ensureCheckboxState(desiredState: boolean, checkbox: Locator): Promise<void> {
+    if (desiredState !== (await this.isEnabled(checkbox))) {
+      await this.switchCheckbox(checkbox);
+    }
+  }
+
+  async isEnabled(checkbox: Locator): Promise<boolean> {
+    await playExpect(checkbox).toBeVisible();
+    const upperElement = checkbox.locator('..').locator('..');
+    const clickableCheckbox = upperElement.getByText('Enabled');
+    return await clickableCheckbox.isVisible();
```

Comment:
I'm not sure I understand what is happening here, why are we looking for some text that says Enabled when the checkbox exposes the method isChecked() which should return current state. Does isChecked() not work for some reason?

Reply:
Yes, it doesn't work, it always returns as checked no matter what the actual status is. I have to check by the label next to it.

Reply:
hey Tamara, I was able to get the rootfyk checkbox to be validated in a recent test I just merged to main, the short of it is that the checkbox can be validated for state, as such checking for Enabled/Disabled text is not required, however to trigger the changing of the state we need to click the parent element, please see https://github.com/containers/podman-desktop/blob/main/tests/playwright/src/specs/podman-machine-tests.spec.ts#L96 and following lines for a model.